lib/fileHandler/index.js

```diff
@@ -104,9 +117,142 @@ module.exports = function (options) {
 
 
 	/*
-  * Download a file from S3. The object passed here should have been one that
-  * was returned via the `upload` function above in a previous workflow step.
-  */
+		A stream variant of upload, requiring a `readStream` property
+		instead of a `file` property
+	*/
+	var streamUpload = function (params) {
+		return when.promise(function (resolve, reject) {
+
+			if (!_.isObject(options.aws)) {
+				return reject(new Error('AWS config not specified. Add a aws.json file in the connector folder, and pass it to falafel.'));
+			}
+
+			if (_.isUndefined(params.readStream) || _.isUndefined(params.readStream.pipe)) {
+				return reject(new Error('The object passed in must contain the property \'readStream\', referecing a read stream.'));
+			}
+
+			var client = knox.createClient({
+				key: options.aws.key,
+				secret: options.aws.secret,
+				bucket: 'workflow-file-uploads',
+				region: 'us-west-2'
+			});
+
+
+			// Generate a random file name
+			var fileName = guid();
+
+			var mimeType = params.contentType || mime.lookup(params.name);
+
+			var req = client.put(fileName, {
+				'Content-Length': params.length,
+				'Content-Type': mimeType,
+			});
+
+			params.readStream.pipe(req);
+
+			req.on('response', function(res){
+				if (res.statusCode !== 200) {
+					reject(new Error(res.statusCode));
```

Comment:
maybe add a bit longer description to the error

Reply:
This is currently consistent with the other file helpers that have been used so far.